src/pages/calendar/components/Section.tsx

```diff
@@ -59,9 +88,39 @@ export function SectionInfo({
   seat,
   meetingTimes,
 }: SectionInfoProps): JSX.Element {
-  const isASYNC = additionalNotes?.indexOf('asynchronous') !== -1;
-  const isSENG = additionalNotes?.indexOf('Reserved for BSENG students') !== -1;
-  const isCSC = additionalNotes?.indexOf('Reserved for students in a Computer Science program') !== -1;
+  const badges: BadgeProps[] = [];
+
+  /* Special cases for async/sync/blended
+   * The UVic notes are not normalized, giving
+   * us inconsistent things to look out for.
+   */
+  if (instructionalMethod === 'online') {
+    if (additionalNotes?.indexOf('mix of “real-time” and asynchronous') !== -1) {
+      badges.push({
+        name: 'Blended',
+        color: 'cyan',
+      });
+    } else if (additionalNotes?.indexOf('fully online and asynchronous') !== -1) {
+      badges.push({
+        name: 'Asynchronous',
+        color: 'cyan',
+      });
+    } else {
+      badges.push({
+        name: 'Synchronous',
+        color: 'cyan',
+      });
+    }
+  }
```

Comment:
Nesting if statements is a bit of a "code-smell" (or _sus_ in modern terms perhaps).
I would either move this into a function or use some guard clauses to clean this up. 
From line 91 to 123 could be moved into a separate function it would help. This also makes testing a lot easier as it will separate your logic from the frontend code that displays the data.